Compiler.cmake

```diff
@@ -11,6 +11,7 @@
 # Input Variables
 # * COMMON_MINIMUM_GCC_VERSION check for a minimum gcc version, default 4.4
 # * COMMON_USE_CXX03 When set, do not enable C++11 language features
+# * COMMON_DISABLE When set, do not set any flags
```

Comment:
If this is to be used generally, then I think it is important that arbitrary flags can be set. At the top level. If we need to build a consistent set of libraries for BBP consumption, then we can agree to use compatible flags, and even have a bbp-official-cpp-flags cmake module. Regarding flag compatibility between dependencies, how do we check / ensure that in the current code WRT installed dependencies?